### ISSUE TITLE
Problem with default handlebar templates in 2.6.200

### ISSUE THREAD

**Issue by sgartner-cityofboise** [ErikEJ/EFCorePowerTools#2304]
### Steps to reproduce the bug

I regenerated the files for a small change to my database and found that it no longer generated the ForeignKey and InverseProperty attributes for the last connecting property in many of the files.

Here's an example.  This is the file generated by the previous version of EF Core PT (same handlebar templates):
```
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace IT.ApplicationSecurityAdmin2.DAL.Models
{
    [Table("AuthenticationType", Schema = "AppSecApplication")]
    public partial class AuthenticationType
    {
        public AuthenticationType()
        {
            Applications = new HashSet<Application>();
        }

        [Key]
        public int AuthenticationTypeId { get; set; }
        [StringLength(50)]
        public string Name { get; set; } = null!;
        public string? Description { get; set; }

        [ForeignKey(nameof(AuthenticationTypeId))]
        [InverseProperty(nameof(Application.AuthenticationTypes))]
        public virtual ICollection<Application> Applications { get; set; }
    }
}
```
Applications has the ForeignKey and InverseProperty attributes.

Here is the new file:
```
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace IT.ApplicationSecurityAdmin2.DAL.Models
{
    [Table("AuthenticationType", Schema = "AppSecApplication")]
    public partial class AuthenticationType
    {
        public AuthenticationType()
        {
            Applications = new HashSet<Application>();
        }

        [Key]
        public int AuthenticationTypeId { get; set; }
        [StringLength(50)]
        public string Name { get; set; } = null!;
        public string? Description { get; set; }

        public virtual ICollection<Application> Applications { get; set; }
    }
}
```

It is identical except that Applications no longer has the proper attributes.  I found this to be true in all classes that ended in a connection property.  If the class ends in a regular property, it has its attributes (like [StringLength()], [Column()], etc.).

If I turn off using handlebar templates then I get the following file, which is correct:
```
// <auto-generated> This file has been auto generated by EF Core Power Tools. </auto-generated>
#nullable enable
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace IT.ApplicationSecurityAdmin2.DAL.Models;

[Table("AuthenticationType", Schema = "AppSecApplication")]
public partial class AuthenticationType
{
    [Key]
    public int AuthenticationTypeId { get; set; }

    [StringLength(50)]
    public string Name { get; set; } = null!;

    public string? Description { get; set; }

    [ForeignKey("AuthenticationTypeId")]
    [InverseProperty("AuthenticationTypes")]
    public virtual ICollection<Application> Applications { get; set; } = new List<Application>();
}
```

I have attached my configuration file.
[efpt.config.json](https://github.com/ErikEJ/EFCorePowerTools/files/15195832/efpt.config.json)

### Provide technical details
- EF Core Power Tools version: 2.6.200
- Exact Visual Studio version: Visual Studio 2022 17.8.5
- Database engine: SQL Server 2022 16.0.1000.6
- EF Core version in use: EF Core 8
- Is Handlebars templates used: yes, the default templates, no changes
- Is T4 templates used: no
- Is a SQL Server .dacpac used: no

**Comment by ErikEJ** [comment]
Have you Switched from EF Core 7 to 8. In that case, please reset the templates.

Why do you need them at all?

Otherwise it could be a template issue  in that case please create an issue in the Handlbars EF Core repo.

**Comment by sgartner-cityofboise** [comment]
I guess I need to move this to a different forum?  I did mention in my notes that I used the most recent handlebar templates.  Your suggestion that I stop using the templates is a tough pill to swallow because then everything changes in the generated files and I have to retest everything for a tiny change.  Initially I was using the templates to solve a problem with the default conversion which I think you have since fixed, but now it's a lot of work to change back.

I don't actually know where the problem is but, since the problem exists in my old handlebar templates and the default handlebar templates, I have to assume it's not in the templates, but in whatever code is processing the templates.
